### PR TITLE
Cleanup dev registry messaging

### PR DESCRIPTION
**Pull request by penalosa**
Cleans up & clarifies dev registry messaging (mostly in response to https://github.com/cloudflare/workers-sdk/issues/7009#issuecomment-2427012598) cc @isaac-mcfadyen 

- Shift from inscrutable red & green dots to `connected`/`not connected` with an explanation logged the first time
Screenshot 2024-11-05 at 13 50 49

- Remove the "Live workers" warning in local mode, where it doesn't apply
---

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required
  - [ ] Not required because:
- Public documentation
  - [ ] TODO (before merge)
  - [x] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/17994
  - [ ] Documentation not necessary because:

### CODE REVIEW

File: packages/wrangler/src/config/index.ts [hunk 1]
```diff
@@ -236,6 +237,7 @@ export function printBindings(
 		local?: boolean;
 	} = {}
 ) {
+	let someConnectionStatus = false;
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/config/index.ts:240]: nit?
```suggestion
	let hasConnectionStatus = false;
```

File: packages/wrangler/src/triggers/deploy.ts [hunk 2]
```diff
@@ -241,7 +241,7 @@ export default async function triggersDeploy(
 	}
 
 	if (config.workflows?.length) {
-		logger.warnOnce("Workflows is currently in open beta.");
+		logger.once.warn("Workflows is currently in open beta.");
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/triggers/deploy.ts:244]: do we need to dbl check this change with the Workflows team?

**Comment by edmundhung** [comment]
Why do we need the concept on connected here? Shouldn't we just make it an error if it is not connected?

File: packages/wrangler/src/triggers/deploy.ts (hunk 2 above)
Comment (penalosa) [reply packages/wrangler/src/triggers/deploy.ts:244]: No—this is the same behaviour, it's just a change in how the `warnOnce()` function is defined

**Comment by penalosa** [comment]
> Why do we need the concept of connected here? Shouldn't we just make it an error if it is not connected?

I'm not sure I understand what you mean here—what should be an error?

File: packages/wrangler/src/config/index.ts [hunk 3]
```diff
@@ -463,14 +466,16 @@ export function printBindings(
...
 					const registryDefinition = context.registry?.[service];
+					hasConnectionStatus = true;
+
 					if (
 						registryDefinition &&
 						(!entrypoint ||
 							registryDefinition.entrypointAddresses?.[entrypoint])
 					) {
-						value = `🟢 ` + value;
+						value = value + " " + chalk.green("connected");
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/config/index.ts:476]: nit: do you think smth like 👇 would better differentiate between the binding and its status?
```suggestion
						value = value + " " + chalk.green("[connected]");
```

**Review by CarmenPopoviciu — APPROVED** [review]

**Comment by isaac-mcfadyen** [comment]
> Cleans up & clarifies dev registry messaging (mostly in response to [#7009 (comment)](https://github.com/cloudflare/workers-sdk/issues/7009#issuecomment-2427012598)) cc @isaac-mcfadyen

Awesome, thank you @penalosa! 😄

File: packages/wrangler/src/config/index.ts (hunk 3 above)
Comment (isaac-mcfadyen) [reply packages/wrangler/src/config/index.ts:476]: +1 on this especially in scenarios when terminal colors are turned off (CI, certain restricted SSH connections or older terminals, etc)

**Pull request merged**